engine/schema/src/com/cloud/storage/dao/StoragePoolTagsDaoImpl.java

```diff
@@ -77,4 +92,71 @@ public void deleteTags(long poolId) {
         txn.commit();
     }
 
+    @Override
+    public List<StoragePoolTagVO> searchByIds(Long... stIds) {
+        final int detailsBatchSize = getDetailsBatchSize();
+
+        // query details by batches
+        List<StoragePoolTagVO> uvList = new ArrayList<StoragePoolTagVO>();
+        int curr_index = 0;
+
+        if (stIds.length > detailsBatchSize) {
```

Comment:
I think that `if` is needed to control the max query size. I agree with the example you provided, but let's say for example that lengthOfStIds is greater that batchSize. If we remove that `if`, we will load pools on lines 100-112 but with a query size greater that batchSize (defined in line 111)